docs/idempotency/README.md

```diff
@@ -0,0 +1,28 @@
+# IdempotencyAnalyzer
+Each of the Stargate persistence implementations (C* 3.11, C* 4.0 and DSE) are providing 
+the IdempotencyAnalyzer component:
+- [IdempotencyAnalyzer C* 3.11]
+- [IdempotencyAnalyzer C* 4.0]
+- [IdempotencyAnalyzer DSE]
+
+Its responsibility is to analyze `CqlStatement`s and infer whether the query is idempotent or not.
+All prepared queries are analyzed.
+If yes, it is safe to retry such a query on the client or/and server-side. 
+If it is not, the query should not be retried because it may result in an inconsistent state of a Database.
+
+## Idempotency rules
```

Comment:
Shouldn't this be javadoc?

Reply:
why? I think that we should have more docs in `readme.md` form that contains consolidated info about this feature. I can also add Javadoc if you think it's needed.

Reply:
Because it's easier to read documentation next to the code you're looking at rather than jumping between code and external docs. Plus the closer docs are to code the more likely it is to stay up to date. Not specific to your PR but in general we're lacking on javadoc which makes it difficult for those not immersed in the project or specific feature to understand what's going on.